### DISCUSSION TITLE
How to use Kong with DNS_RESOLVER?

### DISCUSSION THREAD

**Discussion by kuangshp** [Kong/kong#8092] [Help]
I use Go to write micro-service, use Consul for service discovery, want to use Kong as gateway, use dNS_resolver to read service from Consul, but the request is not working，My dNS_resolver configuration is as follows，172.25.77.238 is the local address of my server
![image](https://user-images.githubusercontent.com/15928680/142571606-8f3434db-d031-47d7-8136-183cd6af55df.png)

The service configuration is as follows
![image](https://user-images.githubusercontent.com/15928680/142571740-bc6804ac-986a-47d5-b9e0-b7ac7eabb8f7.png)

Consul's services are as follows
![image](https://user-images.githubusercontent.com/15928680/142571999-e0664eaf-947b-4edd-8614-9adcb1b23fac.png)

There is no problem using DIA to test user services on the server

```shell
[root@iZ3f5shpfvnl5bZ ~]# dig @172.25.77.238 -p 8600 user_web.service.consul SRV                                  

; <<>> DiG 9.11.26-RedHat-9.11.26-6.el8 <<>> @172.25.77.238 -p 8600 user_web.service.consul SRV
; (1 server found)
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 6563
;; flags: qr aa rd; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 3
;; WARNING: recursion requested but not available

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4096
;; QUESTION SECTION:
;user_web.service.consul.       IN      SRV

;; ANSWER SECTION:
user_web.service.consul. 0      IN      SRV     1 1 33461 ac194dee.addr.dc1.consul.

;; ADDITIONAL SECTION:
ac194dee.addr.dc1.consul. 0     IN      A       172.25.77.238
1.node.dc1.consul.      0       IN      TXT     "consul-network-segment="

;; Query time: 58 msec
;; SERVER: 172.25.77.238#8600(172.25.77.238)
;; WHEN: Fri Nov 19 13:38:07 CST 2021
;; MSG SIZE  rcvd: 155

[root@iZ3f5shpfvnl5bZ ~]# 
```

You can use curl to request something directly

```shell
[root@iZ3f5shpfvnl5bZ ~]# curl localhost:33461/user/api/v1/user                                                   
{"code":0,"message":"success","result":{"data":[],"total":0}}
```

But I can't use Postman locally

![image](https://user-images.githubusercontent.com/15928680/142572273-3ea0d76f-9811-4313-afaa-80b6c07973af.png)

**Comment by mayocream** [comment]
Hi @kuangshp, thank you for reporting this.

Can you provide your running Kong version and the error logs?
You can set `log_level=debug` to print more details for the balancer.
